Formula/c/cody.rb

```diff
@@ -0,0 +1,26 @@
+class Cody < Formula
+  desc "Sourcegraph Cody CLI"
+  homepage "https://sourcegraph.com"
+  url "https://registry.npmjs.org/@sourcegraph/cody/-/cody-5.5.15.tgz"
+  sha256 "2f25357f26e7710b71e4abead3ad9ee3fc5817fbbd6a78cd6726556982bc935a"
+  license "MIT"
+
+  depends_on "node"
+
+  def install
+    ENV["NPM_CONFIG_PREFIX"] = libexec.to_s
+    system "npm", "install", *std_npm_args, "@sourcegraph/cody"
+    rm Dir["#{bin}/*"]
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+
+    (bin/"cody").atomic_write <<~EOS
+      #!/bin/bash
+      export NODE_NO_WARNINGS=1
+      exec "#{libexec}/bin/cody" "$@"
+    EOS
+  end
+
+  test do
+    system bin/"cody", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks @SMillerDev, replaced basic version test with assert test for failed auth due to required credentials.
Please, let me know if I need that I improve something else.